ci/build-all-steps.yml

```diff
@@ -25,40 +25,44 @@ steps:
 - script: npm install
   displayName: npm install
 
-# Slice the tasks (sets TASK_PATTERN)
-- script: node ./ci/set-task-slice.js
-  displayName: Set task slice
+# Filter out unchanged tasks
+- script: node ./ci/filter-tasks.js
+  displayName: Filter out unchanged tasks
 
 # Build
 - script: node make.js build --task "$(task_pattern)"
   displayName: Build
+  condition: ne(variables['numTasks'], 0)
 
 # Only on Windows:
 - ${{ if eq(parameters.os, 'Windows_NT') }}:
 
-  # Stage slice
+  # Stage packages
   - script: node ./ci/stage-slice.js
-    displayName: Stage slice
-    condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'))
+    displayName: Stage package
+    condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'), ne(variables['numTasks'], 0))
 
-  # Publish slice artifact
+  # Publish artifact
   - task: PublishBuildArtifacts@1
```

Comment:
I think we can get rid of this whole step? We publish the packages at the end. This existed as an intermediate step to publish slices so we could then download them again when the slices were done. Since we aren't slicing I think we can just publish once at the end. Also, for non Windows builds we don't need to publish anything.

Reply:
Good call, I missed that we're publishing again at the end.

Reply:
We still probably need something here though as I think about it. Since we're publishing in a different job, nothing is going to persist from this job to that job, right? So the right thing to do here might just be to remove any references/concept of slices in the code, but still publish something here. Otherwise I think we'd have to rebuild in the publish job.

Reply:
I renamed appropriately and changed the download task to only download the first slice.